### PR TITLE
MLHR-1748 Created concrete input and output operators for Redis Store

### PR DESCRIPTION
**Pull request by ishark**
Added test cases for the same.

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/common/FieldInfo.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
+package com.datatorrent.contrib.common;
```
Comment (PramodSSImmaneni) [inline contrib/src/main/java/com/datatorrent/contrib/common/FieldInfo.java]: FieldInfo is now in util

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisKeyValueInputOperator.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  @Override
+  public void beginWindow(long windowId) {
+	System.out.println("Calling begin window again.. ");
+    super.beginWindow(windowId);
+    keysObjectList = new ArrayList<Object>(keys);
+  }
+
+
+  @Override
+  public void emitTuples()
```
Comment (PramodSSImmaneni) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisKeyValueInputOperator.java]: We would like to emit the next available data each time emitTuples is called following the first call

File: contrib/src/main/java/com/datatorrent/contrib/redis/AbstractRedisInputOperator.java [hunk 3]
```diff
@@ -15,18 +15,153 @@
...
-public abstract class AbstractRedisInputOperator<T> extends AbstractKeyValueStoreInputOperator<T, RedisStore>
-{
+public abstract class AbstractRedisInputOperator<T> extends
+    AbstractStoreInputOperator<T, RedisStore> implements CheckpointListener {
+  protected transient List<String> keys = new ArrayList<String>();
+  protected transient String scanOffset;
+  protected transient ScanParams scanParameters;
+  private transient boolean scanComplete;
+  private transient Integer recoveryOffset, backupOffset;
+  private transient int scanCount;
```
Comment (PramodSSImmaneni) [inline contrib/src/main/java/com/datatorrent/contrib/redis/AbstractRedisInputOperator.java]: Is scan count transient?

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisKeyValueInputOperator.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ */
+public class RedisKeyValueInputOperator extends
+    AbstractRedisInputOperator<KeyValPair<String, String>> {
+
+  private List<Object> keysObjectList = new ArrayList<Object>();
+
+  @Override
+  public void beginWindow(long windowId) {
+    super.beginWindow(windowId);
```
Comment (chandnisingh) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisKeyValueInputOperator.java]: There is nothing additional being done so why to override?

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOInputOperator.java [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  @Override
+  public void setup(OperatorContext context) {
+    super.setup(context);
+  }
+
+  public void processFirstTuple(Map<String, String> value)
+      throws ClassNotFoundException {
+    objectClass = Class.forName(getOutputClass());
+
+    System.out.println("In processFirstTuple... ");
```
Comment (chandnisingh) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOInputOperator.java]: Please remove System.out

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOInputOperator.java [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+ * This output adapter Reads from RedisStore stored as <Key, Map>
+ * It outputs a Key value pair <key, POJO> as tuples.
+ * </p>
+ *
+ * @displayName Redis POJO Input Operator
+ * @category Store
+ * @tags output operator, key value
+ *
+ */
+public class RedisPOJOInputOperator extends AbstractRedisInputOperator<KeyValPair<String, Object>> {
```
Comment (chandnisingh) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOInputOperator.java]: Please annotate is as Evolving

File: contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOInputOperator.java [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+            break;
+          default:
+
+            break;
+          }
+        }
+      }
+      return mappedObject;
+    } catch (Exception e) {
+      DTThrowable.rethrow(e);
```
Comment (chandnisingh) [inline contrib/src/main/java/com/datatorrent/contrib/redis/RedisPOJOInputOperator.java]: Can use wrapIfChecked()

File: contrib/src/test/java/com/datatorrent/contrib/redis/RedisPOJOOperatorTest.java [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+      
+      TestClass a = (TestClass)ObjectCollectorModule.resultMap.get("test_abc_in");
+      Assert.assertNotNull(a);
+      Assert.assertEquals("abc", a.stringValue);
+      Assert.assertEquals("1", a.intValue.toString());
+    }
+
+    finally {
+      for (KeyValPair<String, String> entry : CollectorModule.resultMap) {
+      System.out.println("entry is  = " + entry.getKey() + " : " + entry.getValue());
```
Comment (chandnisingh) [inline contrib/src/test/java/com/datatorrent/contrib/redis/RedisPOJOOperatorTest.java]: Please remove System.out

**Comment by chandnisingh** [comment]
Please close this pull request and create a new one against 3.1.0

**Comment by ishark** [comment]
Addressed all the review comments. Will open a pull request against v3.1.9. Closing this one.

**Pull request closed without merging**